### DISCUSSION TITLE
CRUD vs Route duplicate code

### DISCUSSION THREAD

**Discussion by andrewbilham** [fastapi/full-stack-fastapi-template#1231] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
N/A
```

### Description

Good Morning,

When looking at the template it seems the api route files contain the logic that is also in the CRUD file. 

What is the purpose of both the files? 

Sorry if this is very basic, self taught and have deployed and built new tables and route successfully but am not trying to understand in a bit more detail the full project.

### Operating System

Windows

### Operating System Details

Other

### Python Version

3.11

**Comment by YuriiMotov** [comment]
It was probably time when project was under reconstruction..
I'll close this. If you still see duplicated code - please provide examples

**Comment by Jaybird9er** [comment]
I was curious about this as well. For example backend.app.crud contains: 
```
def create_item(*, session: Session, item_in: ItemCreate, owner_id: uuid.UUID) -> Item:
    db_item = Item.model_validate(item_in, update={"owner_id": owner_id})
    session.add(db_item)
    session.commit()
    session.refresh(db_item)
    return db_item
```

While backend.app.routes.items.py contains: 
```
@router.post("/", response_model=ItemPublic)
def create_item(
    *, session: SessionDep, current_user: CurrentUser, item_in: ItemCreate
) -> Any:
    """
    Create new item.
    """
    item = Item.model_validate(item_in, update={"owner_id": current_user.id})
    session.add(item)
    session.commit()
    session.refresh(item)
    return item
```

To maintain a single source of truth, would it not be better to update items.py to this:
```
# ...existing code...
from app import crud
# ...existing code...

@router.post("/", response_model=ItemPublic)
def create_item(
    *, session: SessionDep, current_user: CurrentUser, item_in: ItemCreate
) -> Any:
    """
    Create new item.
    """
    item = crud.create_item(session=session, item_in=item_in, owner_id=current_user.id)
    return item
# ...existing code...
```

I too am largely self-taught, so go easy on me...😅
